lib/node_modules/@stdlib/utils/every-in-by/docs/types/index.d.ts

```diff
@@ -0,0 +1,39 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// Define types for functions/arguments
+type Predicate<T> = ( value: T, key: string, obj: object ) => boolean;
```

Comment:
Please have a look at the type definitions for the predicate function in say `every-by`, which, even though for collections / arrays, should give some indication of how we want to type the `Predicate` function as well as `everyInBy`